src/pkg/artifact/dao/dao.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/internal/orm"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"time"
+)
+
+// DAO is the data access object interface for artifact
+type DAO interface {
+	// GetTotal returns the total count of artifacts according to the query
+	GetTotal(ctx context.Context, query *q.Query) (total int64, err error)
+	// List artifacts according to the query
+	List(ctx context.Context, query *q.Query) (artifacts []*Artifact, err error)
+	// Get the artifact specified by ID
+	Get(ctx context.Context, id int64) (*Artifact, error)
+	// Create the artifact
+	Create(ctx context.Context, artifact *Artifact) (id int64, err error)
+	// Delete the artifact specified by ID
+	Delete(ctx context.Context, id int64) (err error)
+	// UpdatePullTime updates the pull time of the artifact
+	UpdatePullTime(ctx context.Context, artifactID int64, time time.Time) (err error)
```

Comment:
The `Update` can be defined here and the `UpdatePullTime` can be moved to the upper layer?

Reply:
Updated